### PR TITLE
add f90cache caching

### PR DESCRIPTION
**Pull request by junghans**
Related to #3761.

### CODE REVIEW

File: lib/spack/docs/config_yaml.rst [hunk 1]
```diff
@@ -201,3 +201,20 @@ to 4, for example, commands like ``spack install`` will run ``make -j4``
 instead of hogging every core.
 
 To build all software in serial, set ``build_jobs`` to 1.
+
+--------------------
+``f90cache``
+--------------------
+
+When set to ``true`` Spack will use f90cache to cache compiles. This is
+useful specifically un two cases: (1) Use with ``spack setup``, (2)
```
Comment (adamjstewart) [inline lib/spack/docs/config_yaml.rst]: Typo: un -> in

**Comment by junghans** [comment]
I want to get some testing for `f90cache` as well and hence I guess merging `ccache`, which works now, first makes more sense.

**Comment by junghans** [comment]
ping @scheibelp

**Comment by junghans** [comment]
ping @scheibelp

**Comment by junghans** [comment]
Ping @scheibelp

**Comment by junghans** [comment]
ping @scheibelp @tgamblin

**Comment by junghans** [comment]
ping @scheibelp @tgamblin

**Comment by junghans** [comment]
@alalazo for Fortran caching!

**Comment by alalazo** [comment]
Looking at Travis output I think this PR needs a rebase...

**Comment by junghans** [comment]
@alalazo done!

**Comment by alalazo** [comment]
@junghans You know, I just compiled `cp2k` a dozen times. This would have been useful!

**Comment by junghans** [comment]
@alalazo no idea why `f90cache` fails, maybe @muellermichel knows.

**Comment by junghans** [comment]
gcc-9 support was added in #11838.

**Comment by adamjstewart** [comment]
@junghans is this PR still a work in progress or can we close it?

**Pull request closed without merging**